src/lua/utils.c

```diff
@@ -147,11 +147,19 @@ luaT_pushdatetime(struct lua_State *L, const struct datetime *dt)
 }
 
 struct interval *
-luaT_pushinterval(struct lua_State *L)
+luaT_newinterval(struct lua_State *L)
 {
 	return luaL_pushcdata(L, CTID_INTERVAL);
 }
 
+struct interval *
+luaT_pushinterval(struct lua_State *L, const struct interval *ivt)
+{
+	struct interval *res = luaT_newinterval(L);
+	memcpy(res, ivt, sizeof(struct interval));
```

Comment:
Sorry for nitpicking, but it's usually `itv`, not `ivt`.

Reply:
Ouch :) Fixed.

*Please, note that I'm working on the alternative implementation of PR #7428 and this PR will likely be changed a bit. So I don't expect reviews here ATM.*

Reply:
*(Now it is ready for review again. Please, note that the first commit is from PR #7497 and is not object to review here.)*

Reply:
Hm, I lost this change somehow. Sorry, fixed again.